### DISCUSSION TITLE
status of LFS support

### DISCUSSION THREAD

**Discussion by ctanis** [microsoft/git#624] [Q&A]
What is the status of LFS+sparse-checkout support with GVFS via Scalar on Azure?  
I get the impression this is a common scenario, but am concerned about `git lfs fsck` errors after a sparse checkout, as well as other `git lfs` operations populating the work directory with undesired files.

It seems like a usable situation despite these errors, but am concered about long-term reliability of this setup.  The LFS guys say all flavors of VFS are unsupported and probably broken.  

Any thoughts on this topic would be most welcome.

Thanks,
Craig

**Comment by dscho** [comment]
> [I] am concerned about git lfs fsck errors after a sparse checkout [...]

As far as Git LFS is concerned, GVFS should not be used, it is either/or. But since you mentioned Scalar, I figure that you're not really using VFS for Git at all, but regular Git after a `scalar clone`. On Git LFS' side, it does not even matter whether it was Scalar or not that set up this worktree, but it matters that the checkout is sparse. I don't know about the status of `git lfs fsck` regarding sparse checkouts but could imagine that this command does not know about sparse checkouts and therefore is confused by the absence of the file(s).

Having said that, the fundamental principle of Git LFS is that it works via Git's own clean/smudge filters which _do_ work as intended with sparse checkouts. Therefore I would guess that all your scenario needs is for `git lfs fsck` to learn about sparse checkouts by ignoring the intended absence of files that are unmatched by the sparse definition.

**Comment by ctanis** [comment]
Thanks for the response.. it's clear that I'm a bit fuzzy on some of the Windows-specific aspects of this. 

Here's what I thought -- any corrections would be most welcome:  GVFS is the Azure-specific protocol that microsoft/git  provides, and this fork (rather than "the official git") was the appropriate version to use if you were working with Azure so that git/scalar could provide sparse-checkout properly.  I assumed this was unrelated to whether or not LFS was used.  "VFS" refers to the filesystem-based implementation of the original scalar functionality and would be an alternative to LFS if it weren't obsolete.

**Reply by derrickstolee** [reply]
Using the GVFS protocol at the same time as LFS might lead to some silliness as the LFS pointer files must be downloaded via the GVFS protocol before they can be accessed and sent to the `git-lfs` client for downloading the actual content. But it should all work.
